types/src/key.rs

```diff
@@ -319,12 +341,33 @@ impl Key {
         }
     }
 
+    /// Returns a reference to the inner [`LocalAddr`] if `self` is of type [`Key::Local`],
+    /// otherwise returns `None`.
+    pub fn as_local(&self) -> Option<&LocalAddr> {
+        match self {
+            Key::Dictionary(v) => Some(v),
+            _ => None,
+        }
+    }
+
     /// Casts a [`Key::URef`] to a [`Key::Hash`]
     pub fn uref_to_hash(&self) -> Option<Key> {
         let uref = self.as_uref()?;
         let addr = uref.addr();
         Some(Key::Hash(addr))
     }
+
+    /// Creates a new [`Key::Local`] variant based on a `uref` and a `key` bytes.
+    pub fn dictionary(uref: URef, key: &[u8]) -> Key {
+        // NOTE: Expect below is safe because the length passed is supported.
+        let mut hasher = VarBlake2b::new(BLAKE2B_DIGEST_LENGTH).expect("should create hasher");
+        hasher.update(uref.addr().as_ref());
+        hasher.update(key);
+        // NOTE: Assumed safe as size of `HashAddr` equals to the output provided by hasher.
+        let mut local_addr = HashAddr::default();
+        hasher.finalize_variable(|hash| local_addr.clone_from_slice(hash));
```

Comment:
I would prefer to use the `Blake2bHash` and its `new` method here, even if we are "unwrapping" from the `Blake2bHash` newtype itself when we write the dictionary key.  That newtype was designed to abstract over this operation, and prevents us from duplicating (and/or getting wrong) the hashing logic.

Reply:
Unfortunately `Blake2bHash` is in ./execution_engine, but this code is in ./types. I could use the `account::blake2b` function, or do we want to move the `Blake2bHash` struct into the `types` crate as part of this PR?

Reply:
Tragic, let's keep as-is for now